Cannot build the libs once they are installed

I am building `capnproto` using VS2019 (MSVC), `cmake` and `Ninja`. I am specifying `CMAKE_INSTALL_PREFIX` on command line for two reasons: 1) instruct the toolchain where to look for the additional dependencies (e.g. `zlib`), 2) have the `capnproto` libs installed there.

For example the command line options for the debug build look like this:
```
cmake.exe  -G "Ninja" ^
-DCMAKE_BUILD_TYPE:STRING="Debug" ^
-DCMAKE_INSTALL_PREFIX:PATH="C:\Dev\crtdll" ^
-DCMAKE_CXX_STANDARD=17 ^
-DCMAKE_EXPORT_COMPILE_COMMANDS=ON ^
-DCMAKE_MSVC_RUNTIME_LIBRARY="MultiThreaded$<$<CONFIG:Debug>:Debug>DLL" ^
"C:\Work_OSS\capnproto" 2>&1"
```

This builds fine as long as I do not install the library. Once the library is installed (into `C:\Dev\crtdll`) the build fails at the first target:
```
>------ Build All started: Project: capnproto, Configuration: x64-DLL-Debug ------
  [1/185] C:\PROGRA~2\MICROS~4\2019\COMMUN~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP  -IC:\Dev\crtdll\include -I..\..\..\c++\src /DWIN32 /D_WINDOWS /W3 /GR /EHsc /MDd /Zi /Ob0 /Od /RTC1   -D KJ_HAS_ZLIB=1 /wo4503 -std:c++17 /showIncludes /Foc++\src\kj\CMakeFiles\kj-tests.dir\string-test.c++.obj /Fdc++\src\kj\CMakeFiles\kj-tests.dir\ /FS -c ..\..\..\c++\src\kj\string-test.c++
  FAILED: c++/src/kj/CMakeFiles/kj-tests.dir/string-test.c++.obj 
  C:\PROGRA~2\MICROS~4\2019\COMMUN~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP  -IC:\Dev\crtdll\include -I..\..\..\c++\src /DWIN32 /D_WINDOWS /W3 /GR /EHsc /MDd /Zi /Ob0 /Od /RTC1   -D KJ_HAS_ZLIB=1 /wo4503 -std:c++17 /showIncludes /Foc++\src\kj\CMakeFiles\kj-tests.dir\string-test.c++.obj /Fdc++\src\kj\CMakeFiles\kj-tests.dir\ /FS -c ..\..\..\c++\src\kj\string-test.c++
  string-test.c++
C:\Dev\crtdll\include\kj\common.h(288): error C2572: 'kj::_::inlineRequireFailure': redefinition of default argument: parameter 1
  C:\Work_OSS\capnproto\c++\src\kj\common.h(288): note: see declaration of 'kj::_::inlineRequireFailure'
C:\Dev\crtdll\include\kj\common.h(414): error C2953: 'kj::NoInfer_': class template has already been defined
  C:\Dev\crtdll\include\kj\common.h(414): note: see declaration of 'kj::NoInfer_'
C:\Dev\crtdll\include\kj\common.h(415): error C2386: 'NoInfer': a symbol with this name already exists in the current scope
C:\Dev\crtdll\include\kj\common.h(419): error C2953: 'kj::RemoveConst_': class template has already been defined
  C:\Dev\crtdll\include\kj\common.h(419): note: see declaration of 'kj::RemoveConst_'
C:\Dev\crtdll\include\kj\common.h(420): error C2953: 'kj::RemoveConst_': class template has already been defined
  C:\Dev\crtdll\include\kj\common.h(420): note: see declaration of 'kj::RemoveConst_'
```
I guess the problem is that the lib implementation uses "double quote" and "angle-bracket" notation at the same time (to include the same header file), even in one file like this (snippet from `message.h`):
```
#pragma once

#include <kj/common.h>
#include <kj/memory.h>
#include <kj/mutex.h>
#include <kj/debug.h>
#include <kj/vector.h>
#include "common.h"
#include "layout.h"
#include "any.h"
```
This leads to compiler including the same file from two different locations and apparently `#pragma once` is not enough to recognize that. Once I delete the header files from the install path, it starts building fine. 

Apart from the obvious problem quoted above, I believe the problem may manifest at some other includes as well, as it seems that many are including `"common.h"`, but to which one it will resolve (the global or local one) will depend on where exactly the file including this resides relative to `common.h`.

--- a/c++/src/kj/CMakeLists.txt
+++ b/c++/src/kj/CMakeLists.txt
@@ -207,8 +207,14 @@ if(NOT CAPNP_LITE)
   find_package(ZLIB)
   if(ZLIB_FOUND)
     add_definitions(-D KJ_HAS_ZLIB=1)
-    include_directories(${ZLIB_INCLUDE_DIRS})
     target_link_libraries(kj-gzip PUBLIC kj-async kj ${ZLIB_LIBRARIES})
+    # The interface include directories inherited from `kj` lib are going to be the last
+    # in the target include directories order, but we want them to be the first to avoid
+    # including from the ZLIB (possibly) global include path first, as it may contain
+    # capnproto includes as well.
+    # NOTE: kj-async does not export any interface properties so we can ignore it here
+    get_target_property(_KJ_INTERFACE_INCLUDE_DIRECTORIES kj INTERFACE_INCLUDE_DIRECTORIES)
+    target_include_directories(kj-gzip PUBLIC "${_KJ_INTERFACE_INCLUDE_DIRECTORIES};${ZLIB_INCLUDE_DIRS}")
   endif()
 
   # Ensure the library has a version set to match autotools build

--- a/c++/src/kj/CMakeLists.txt
+++ b/c++/src/kj/CMakeLists.txt
@@ -207,14 +207,7 @@ if(NOT CAPNP_LITE)
   find_package(ZLIB)
   if(ZLIB_FOUND)
     add_definitions(-D KJ_HAS_ZLIB=1)
-    target_link_libraries(kj-gzip PUBLIC kj-async kj ${ZLIB_LIBRARIES})
-    # The interface include directories inherited from `kj` lib are going to be the last
-    # in the target include directories order, but we want them to be the first to avoid
-    # including from the ZLIB (possibly) global include path first, as it may contain
-    # capnproto includes as well.
-    # NOTE: kj-async does not export any interface properties so we can ignore it here
-    get_target_property(_KJ_INTERFACE_INCLUDE_DIRECTORIES kj INTERFACE_INCLUDE_DIRECTORIES)
-    target_include_directories(kj-gzip PUBLIC "${_KJ_INTERFACE_INCLUDE_DIRECTORIES};${ZLIB_INCLUDE_DIRS}")
+    target_link_libraries(kj-gzip PUBLIC kj-async kj ZLIB::ZLIB)
   endif()
 
   # Ensure the library has a version set to match autotools build

